onnx/defs/nn/defs.cc

```diff
@@ -264,7 +264,14 @@ and computes the output.)DOC";
                         "The shape of the convolution kernel.",
                          AttributeProto::INTS, OPTIONAL);
             schema.Attr("output_shape",
-                        "The shape of the output.",
+                        "The shape of the output."
+                        " output_shape[i] = stride[i] * (input_size[i] - 1) + output_padding[i] +"
+                        " kernel_shape[i] - pads[start_i] - pads[end_i]",
+                        AttributeProto::INTS, OPTIONAL);
+            schema.Attr("output_padding",
+                        "The zero-padding added to one side of the output."
+                        " This is also called adjs/adjustment in some frameworks."
+                        " If output_shape is set, this attribute will be ignored.",
```

Comment:
OK, given that we're in the "non BC-breaking changes are OK to edit directly" regime, it would be nice to mention a date when new attributes are added to make it clear that they're less likely to be implemented by backends. Additionally, since this is an optional attribute without a default, it would be good to say what its semantics are when it's not set. (In this case, no padding; maybe this is reasonably clear and not necessary.)